x-pack/plugins/ml/public/application/components/data_grid/common.ts

```diff
@@ -180,69 +185,44 @@ export const getFeatureImportance = (
   mlResultsField: string,
   isClassTypeBoolean = false
 ): FeatureImportance[] => {
-  const featureNames: string[] | undefined =
-    row[`${mlResultsField}.feature_importance.feature_name`];
-  const classNames: string[] | undefined =
-    row[`${mlResultsField}.feature_importance.classes.class_name`];
-  const classImportance: number[] | undefined =
-    row[`${mlResultsField}.feature_importance.classes.importance`];
-
-  if (featureNames === undefined) {
-    return [];
-  }
-
-  // return object structure for classification job
-  if (classNames !== undefined && classImportance !== undefined) {
-    const overallClassNames = classNames?.slice(0, classNames.length / featureNames.length);
-
-    return featureNames.map((fName, index) => {
-      const offset = overallClassNames.length * index;
-      const featureClassImportance = classImportance.slice(
-        offset,
-        offset + overallClassNames.length
-      );
-      return {
-        feature_name: fName,
-        classes: overallClassNames.map((fClassName, fIndex) => {
+  const featureImportance: Array<{
+    feature_name: string[];
+    classes?: Array<{ class_name: FeatureImportanceClassName[]; importance: number[] }>;
+    importance?: number;
+  }> = row[`${mlResultsField}.feature_importance`];
+  if (featureImportance === undefined) return [];
+
+  return featureImportance.map((fi) => ({
+    feature_name: Array.isArray(fi.feature_name) ? fi.feature_name[0] : fi.feature_name,
+    classes: Array.isArray(fi.classes)
+      ? fi.classes.map((c) => {
+          const processedClass = getProcessedFields(c);
           return {
-            class_name: getClassName(fClassName, isClassTypeBoolean),
-            importance: featureClassImportance[fIndex],
+            importance: processedClass.importance,
+            class_name: getClassName(processedClass.class_name, isClassTypeBoolean),
           };
-        }),
-      };
-    });
-  }
-
-  // return object structure for regression job
-  const importance: number[] = row[`${mlResultsField}.feature_importance.importance`];
-  return featureNames.map((fName, index) => ({
-    feature_name: fName,
-    importance: importance[index],
+        })
+      : fi.classes,
+    importance: Array.isArray(fi.importance) ? fi.importance[0] : fi.importance,
```

Comment:
From the type, it looks like 'importance' is a number. If it can also be an array of numbers it might be worth updating the type.